t/90h2olog.t

```diff
@@ -8,6 +8,9 @@ use Test::More;
 use JSON;
 use t::Util;
 
+plan skip_all => "DTRACE_TESTS=0 is set to skip this file"
+    if defined($ENV{DTRACE_TESTS}) && !$ENV{DTRACE_TESTS};
```

Comment:
```suggestion
    unless $ENV{DTRACE_TESTS};
```
Maybe this is sufficient?

Reply:
I'd like to run this test file if `DTRACE_TESTS` is not set. That is, my intension is:

* `DTRACE_TESTS=1` to force dtrace tests
* `DTRACE_TESTS=0` to skip dtrace tests
* no `DTRACE_TESTS` specified  to run dtrace tests if prerequisites are met

Reply:
I'm not sure if that's consistent with how we `DTRACE_TESTS`. Can I ask why you think something like that is necessary?

Rereading existing code, I have several feelings:
* It is a mistake to refer to `DTRACE_TESTS` in h2olog.t - it should be calling `check_dtrace_availability` instead.
* The rule around `DTRACE_TESTS` is probably unneed. In `check_dtrace_availability`, probably all we need to do is check prerequisites, including the version check of `bpftrace`. If `bpftrace` is too old, we should skip tests. Otherwise, we should run them. There is no need to have a flag that allows a user to specify whether some test should be run.

WDYT?

Reply:
Discussed offline with @gfx, agreed that t/90h2olog.t should simply depend on `check_dtrace_availability`.

We will keep `check_dtrace_availability` as is, for the time being. The complexity of running dtrace tests is not only on the linux side; on macOSX, it requires a boot parameter change too.

Reply:
I could remove the changes in test files after all. `check_dtrace_availability()` has a bit different purpose, which checks tracer tools such as `dtrace(1)` or `bpftrace(1)`. 

I'd like to refactor around them (e.g. coming up with a new utility function that are shared in all the dtrace / h2olog tests), but will do in a separate PR.